### PR TITLE
Add Amazon Managed Prometheus managed scraper resource

### PR DESCRIPTION
**Pull request by bonclay7**
### Description
This PR adds the newly announced managed collector for Amazon Managed Service for Prometheus
https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-collector.html

In another PR, I'll submit a data source for fetching a service managed default scrape configuration to use with this new resource

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/34733.
Closes https://github.com/hashicorp/terraform-provider-aws/issues/34868.

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccAMPScraper PKG=amp 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 20 -run='TestAccAMPScraper'  -timeout 360m
=== RUN   TestAccAMPScraper_basic
=== PAUSE TestAccAMPScraper_basic
=== RUN   TestAccAMPScraper_disappears
=== PAUSE TestAccAMPScraper_disappears
=== RUN   TestAccAMPScraper_tags
=== PAUSE TestAccAMPScraper_tags
=== RUN   TestAccAMPScraper_alias
=== PAUSE TestAccAMPScraper_alias
=== CONT  TestAccAMPScraper_basic
=== CONT  TestAccAMPScraper_tags
=== CONT  TestAccAMPScraper_alias
=== CONT  TestAccAMPScraper_disappears
--- PASS: TestAccAMPScraper_disappears (2692.16s)
--- PASS: TestAccAMPScraper_basic (2733.34s)
--- PASS: TestAccAMPScraper_alias (2945.51s)
--- PASS: TestAccAMPScraper_tags (2981.60s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	2985.308s
...
```

### CODE REVIEW

File: internal/service/amp/scraper.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+package amp
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
```
[the remark this answers is not in the document]
Comment (bonclay7) [reply internal/service/amp/scraper.go:12]: Also, for the same service can we have both v2 and v1 client or we need to migrate all AMP to v2 before?

[the remark this answers is not in the document]
Comment (bonclay7) [reply internal/service/amp/scraper.go:12]: I saw `names_data.csv` + make gen can help, just need to figure tags with v2

[the remark this answers is not in the document]
Comment (bonclay7) [reply internal/service/amp/scraper.go]: Figured this out!

**Comment by bonclay7** [comment]
Hi @justinretzolk, any feedback on this PR?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccAMPWorkspace_\|TestAccAMPWorkspaceDataSource_\|TestAccAMPWorkspacesDataSource_' PKG=amp ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 2  -run=TestAccAMPWorkspace_\|TestAccAMPWorkspaceDataSource_\|TestAccAMPWorkspacesDataSource_ -timeout 360m
=== RUN   TestAccAMPWorkspaceDataSource_basic
--- PASS: TestAccAMPWorkspaceDataSource_basic (17.65s)
=== RUN   TestAccAMPWorkspace_basic
=== PAUSE TestAccAMPWorkspace_basic
=== RUN   TestAccAMPWorkspace_disappears
=== PAUSE TestAccAMPWorkspace_disappears
=== RUN   TestAccAMPWorkspace_kms
=== PAUSE TestAccAMPWorkspace_kms
=== RUN   TestAccAMPWorkspace_tags
=== PAUSE TestAccAMPWorkspace_tags
=== RUN   TestAccAMPWorkspace_alias
=== PAUSE TestAccAMPWorkspace_alias
=== RUN   TestAccAMPWorkspace_loggingConfiguration
=== PAUSE TestAccAMPWorkspace_loggingConfiguration
=== RUN   TestAccAMPWorkspacesDataSource_basic
--- PASS: TestAccAMPWorkspacesDataSource_basic (26.43s)
=== RUN   TestAccAMPWorkspacesDataSource_aliasPrefix
--- PASS: TestAccAMPWorkspacesDataSource_aliasPrefix (16.51s)
=== CONT  TestAccAMPWorkspace_basic
=== CONT  TestAccAMPWorkspace_tags
--- PASS: TestAccAMPWorkspace_basic (22.12s)
=== CONT  TestAccAMPWorkspace_loggingConfiguration
--- PASS: TestAccAMPWorkspace_tags (46.48s)
=== CONT  TestAccAMPWorkspace_kms
--- PASS: TestAccAMPWorkspace_kms (19.55s)
=== CONT  TestAccAMPWorkspace_alias
--- PASS: TestAccAMPWorkspace_loggingConfiguration (64.99s)
=== CONT  TestAccAMPWorkspace_disappears
--- PASS: TestAccAMPWorkspace_disappears (17.72s)
--- PASS: TestAccAMPWorkspace_alias (58.52s)
PASS
```
```console
% make testacc TESTARGS='-run=TestAccAMPAlertManagerDefinition_\|TestAccAMPRuleGroupNamespace_' PKG=amp ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 2  -run=TestAccAMPAlertManagerDefinition_\|TestAccAMPRuleGroupNamespace_ -timeout 360m
=== RUN   TestAccAMPAlertManagerDefinition_basic
=== PAUSE TestAccAMPAlertManagerDefinition_basic
=== RUN   TestAccAMPAlertManagerDefinition_disappears
=== PAUSE TestAccAMPAlertManagerDefinition_disappears
=== RUN   TestAccAMPRuleGroupNamespace_basic
=== PAUSE TestAccAMPRuleGroupNamespace_basic
=== RUN   TestAccAMPRuleGroupNamespace_disappears
=== PAUSE TestAccAMPRuleGroupNamespace_disappears
=== CONT  TestAccAMPAlertManagerDefinition_basic
=== CONT  TestAccAMPRuleGroupNamespace_basic
--- PASS: TestAccAMPRuleGroupNamespace_basic (233.29s)
=== CONT  TestAccAMPRuleGroupNamespace_disappears
--- PASS: TestAccAMPAlertManagerDefinition_basic (267.71s)
=== CONT  TestAccAMPAlertManagerDefinition_disappears
--- PASS: TestAccAMPAlertManagerDefinition_disappears (117.33s)
--- PASS: TestAccAMPRuleGroupNamespace_disappears (152.74s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	391.482s
```
```console
% make testacc TESTARGS='-run=TestAccAMPScraper_basic' PKG=amp
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 20  -run=TestAccAMPScraper_basic -timeout 360m
=== RUN   TestAccAMPScraper_basic
=== PAUSE TestAccAMPScraper_basic
=== CONT  TestAccAMPScraper_basic
--- PASS: TestAccAMPScraper_basic (2227.37s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	2232.542s
```
```console
% make testacc TESTARGS='-run=TestAccAMPScraper_alias\|TestAccAMPScraper_securityGroups' PKG=amp ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 2  -run=TestAccAMPScraper_alias\|TestAccAMPScraper_securityGroups -timeout 360m
=== RUN   TestAccAMPScraper_alias
=== PAUSE TestAccAMPScraper_alias
=== RUN   TestAccAMPScraper_securityGroups
=== PAUSE TestAccAMPScraper_securityGroups
=== CONT  TestAccAMPScraper_alias
=== CONT  TestAccAMPScraper_securityGroups
--- PASS: TestAccAMPScraper_alias (2286.95s)
--- PASS: TestAccAMPScraper_securityGroups (2306.16s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	2311.988s
```
```console
% make testacc TESTARGS='-run=TestAccAMPScraper_disappears\|TestAccAMPScraper_tags' PKG=amp ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 2  -run=TestAccAMPScraper_disappears\|TestAccAMPScraper_tags -timeout 360m
=== RUN   TestAccAMPScraper_disappears
=== PAUSE TestAccAMPScraper_disappears
=== RUN   TestAccAMPScraper_tags
=== PAUSE TestAccAMPScraper_tags
=== CONT  TestAccAMPScraper_disappears
=== CONT  TestAccAMPScraper_tags
--- PASS: TestAccAMPScraper_disappears (2167.50s)
--- PASS: TestAccAMPScraper_tags (2190.34s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	2195.614s
```

**Comment by ewbankkit** [comment]
@bonclay7 Thanks for the contribution :tada: :clap:.

**Pull request merged**